include/fplus/numeric.h

```diff
@@ -135,6 +135,107 @@ X int_power(X base, X exp)
     return base * int_power(base, exp - 1);
 }
 
+template <typename... Conds>
+struct and_ : std::true_type
```

Comment:
Does that mean that `and_<>` is true?  In other words, I would expect `and` not to be defined for no conditions at all.  That can be expressed by just providing a forward declaration for `and_`, and supplying a case for a singleton:

    template<typename... Conds> struct and_; // fwd declaration only -> compile error for no conditions
    template<typename Cond> struct and_: public Cond {};
    template<typename Cond, typename Conds...> // as defined